internal/plugin6/grpc_provider.go

```diff
@@ -138,6 +119,18 @@ func (p *GRPCProvider) GetProviderSchema(ctx context.Context) (resp providers.Ge
 		return resp
 	}
 
+	// Mark that this instance of the provider has fetched schemas
+	p.hasFetchedSchemas = true
```

Comment:
It might be nice if we could move this kind of logic up to the SchemaCache level, instead of it being a boolean property on everything. This way there's a clearer seperation of concerns.

Maybe I'm mixing things up here: but I think what you want to track here is that the provider has fetched the schema, is this technically the same as "Does the schema exist in the cache?"

Reply:
I also can't really see where this is consumed, so maybe we could just drop this?

Reply:
This is keeping the `GetProviderSchemaOptional=false` case from the original code.  I removed some comments that should be re-added into the new structure.  This is independent from the SchemaCache level as it tracks if this *instance* has fetched the schema for legacy providers.